builder/builder.go

```diff
@@ -96,10 +94,13 @@ type ImageCache interface {
 type Image interface {
 	ImageID() string
 	RunConfig() *container.Config
+	MarshalJSON() ([]byte, error)
+	NewChild(child image.ChildConfig) *image.Image
```

Comment:
I'm happy to remove this from the `Image` interface, I think that's a good change.

Isn't there a parent/child relationship that is established by the history?

I'm not really clear how this becomes `AddHistory()` though. In addition to history, the parent image is used for:
* `img.RootFS`
* `img.OSFeatures` / `img.OSVersion`

OSFeature/Version doesn't seem to be used for much. Do you think we could drop those as well?

`Image.RootFS` would have to be exposed somewhere if we don't pass in both images to this new function.